exasol_script_languages_container_tool/lib/utils/logging_redirection.py

```diff
@@ -1,45 +1,59 @@
 import sys
+from contextlib import contextmanager
 
 from exasol_integration_test_docker_environment.lib.base.base_task import BaseTask
 
 
+def _get_log_path(task: BaseTask):
+    return f'{task.get_log_path()}/exaslct.log'
+
+
 class _Tee(object):
-    # Helper object which redirects calls to stdout/stderr to itself, prints to a given file and to real stdout/stderr
-    def __init__(self, name, mode):
-        self.file = open(name, mode)
+    """
+    Helper object which redirects calls to stdout/stderr to itself, prints to a given file and to real stdout/stderr
+    """
+
+    def __init__(self):
+        self.is_open = False
+
+    def open(self, task, mode):
+        self.log_file_path = _get_log_path(task)
+        self.file = open(self.log_file_path, mode)
         self.stdout = sys.stdout
         self.stderr = sys.stderr
         sys.stdout = self
         sys.stderr = self
+        self.is_open = True
 
-    def get_file(self):
-        return self.file
-
-    def __del__(self):
-        sys.stdout = self.stdout
-        sys.stderr = self.stderr
-        self.file.close()
+    def close(self):
+        if self.is_open:
+            sys.stdout = self.stdout
+            sys.stderr = self.stderr
+            self.file.close()
+            print(f'log can be found at:{self.log_file_path}')
 
     def write(self, data):
-        self.file.write(data)
-        self.stdout.write(data)
-
-    def flush(self):
-        self.file.flush()
+        if self.is_open:
+            self.file.write(data)
+            self.stdout.write(data)
+        else:
+            raise RuntimeError("_Tee object used but not initialized.")
 
 
+@contextmanager
 def log_redirector_task_creator_wrapper(task_creator):
     # Provides redirection of ALL logs to exaslct.log in the tasks job directory
 
+    _tee = _Tee()
+
     # Parameters:
     #   task_creator (function): creator function which creates task
     def create_task_creator_wrapper():
         task = task_creator()
-        tee = _Tee(get_log_path(task), "w")
+        _tee.open(task, "w")
         return task
 
-    return create_task_creator_wrapper
-
-
-def get_log_path(task: BaseTask):
-    return f'{task.get_log_path()}/exaslct.log'
+    try:
+        yield create_task_creator_wrapper
+    finally:
+        _tee.close()
```

Comment:
why not support the context manager as part of the Tee object or provide it as context manager in general?

Reply:
As discussed offline, transforming the Tee object into a ContextManager is not feasible because of the indirections of the task creation.